app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -16,16 +16,161 @@
 
 package com.google.droidjump;
 
+import static androidx.navigation.Navigation.findNavController;
+import android.annotation.SuppressLint;
 import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
 import android.view.SurfaceView;
 
 public class GameView extends SurfaceView implements Runnable {
 
+    private boolean isPlaying;
+    private int screenX;
+    private int screenY;
+    private int screenMargin;
+    private SurfaceHolder surfaceHolder;
+    private Droid droid;
+    private int timePoint;
+    private Thread thread;
+    private int levelTimePoints;
+    private int levelSpeed;
+
     public GameView(Context context) {
         super(context);
+        isPlaying = false;
+    }
+
+    public GameView(Context context, int screenX, int screenY, boolean isPlaying) {
+        super(context);
+        receiveLevelDetails();
+        timePoint = 0;
+        surfaceHolder = getHolder();
+        this.screenX = screenX;
+        this.screenY = screenY;
+        this.isPlaying = isPlaying;
+
+        // Margin in px
+        screenMargin = (int) getResources().getDimension(R.dimen.fab_margin);
+
+        // Create droid
+        droid = new Droid(screenMargin, screenY - screenMargin, levelSpeed, getResources());
+        droid.setDroidJumpHeight(getResources());
     }
 
     @Override
     public void run() {
+        while (isPlaying) {
+            update();
+            draw();
+            sleep();
+            timePoint++;
+        }
+    }
+
+    private void receiveLevelDetails() {
+        // TODO: Serialize current level data and put it in some container
+        levelTimePoints = 200;
+        levelSpeed = 100;
+    }
+
+    public void update() {
+        // TODO: Check if time point is in level data and add data to some container, than move
+        //  it to left
+        updateDroidCoordinates();
+
+        // Level Finishing
+        if (timePoint == levelTimePoints) {
+            winGame();
+        }
+    }
+
+    private void updateDroidCoordinates() {
+        if (droid.isJumping()) {
+            if (droid.getY() < droid.getInitialY() - droid.getJumpHeight()) {
+                droid.setJumping(false);
```

Comment:
Should we also change animation frame if droid is no longer jumping?

Reply:
We need to change animation after droid land the ground.